packages/www/components/Dashboard/StreamDetails/StreamHealthTab.tsx

```diff
@@ -76,6 +95,24 @@ const StreamHealthTab = ({ stream, streamHealth, invalidateStream }) => {
     };
   }, [getIngestRate, info]);
 
+  useEffect(() => {
+    const multistreamBitrateSec = streamHealth?.metrics?.MultistreamBitrateSec;
+    if (multistreamBitrateSec) {
+      const data: MultistreamChartType = {};
+      multistreamBitrateSec?.forEach((item, i) => {
+        const { dimensions, last } = item;
+        if (i === 0) {
+          data.name = Math.round((last[0] - startTime) / 1000);
+        }
+        data[`${dimensions.targetName} (${dimensions.targetProfile})`] =
+          Math.round(last[1] / 1000);
+      });
+
+      const newData = [...multiDataChart, data].slice(-(maxItems + 10));
+      setMultiDataChart(newData);
```

Comment:
I believe it's more appropriate to send a function to the `setMultiDataChart` that receives the current value and returns the new value. This will avoid any closure hell that could happen here. Like this: https://github.com/livepeer/livepeer-com/blob/59d54adb8c5265b1236fb502429ae75ff4277a1f/packages/www/components/Dashboard/Logger/index.tsx#L152